### ISSUE TITLE
curl fails with localhost - could not resolve

### ISSUE THREAD

**Issue by AndreFabris** [simplepie/simplepie#599]
Hi I am testing locally, and my host is
appsforce.back
+error: "cURL error 6: Could not resolve host: appsforce.back%2Fmyblog%2F
It seems here:
+feed_url: "http://appsforce.back%2Fmyblog%2F/?feed=atom#"
+permanent_url: "http://appsforce.back%2Fmyblog%2F/?feed=atom#"

That this plugin adds http:// which will not work for my local testing

**Comment by AndreFabris** [comment]
In Misc.php you have fix_protocol function with this:
if ($parsed['scheme'] !== '' && $parsed['scheme'] !== 'http' && $parsed['scheme'] !== 'https')
		{
			return SimplePie_Misc::fix_protocol(SimplePie_Misc::compress_parse_url('http', $parsed['authority'], $parsed['path'], $parsed['query'], $parsed['fragment']), $http);
		}
This forces the http in my case

**Comment by JanPetterMG** [comment]
`appsforce.back%2Fmyblog%2F` is not an valid URL.
You need to `urldecode` it so it becomes `appsforce.back/myblog/`. But FYI that's not an valid URL either, the protocol is missing, that's why Simplepie "fixes" it for you.

What protocol are you using if it's not `http`/`https` then?
I'm pretty sure `http://appsforce.back/myblog/` is the correct URL.

**Comment by AndreFabris** [comment]
Hello

this bit of code
public function set_feed_url($url)
	{
		$this->multifeed_url = array();
		if (is_array($url))
		{
			foreach ($url as $value)
			{
				$this->multifeed_url[] = $this->registry->call('Misc', 'fix_protocol', array($value, 1));
			}
		}
		else
		{
			$this->feed_url = $this->registry->call('Misc', 'fix_protocol', array($url, 1));
			$this->permanent_url = $this->feed_url;
		}
	}

changes the url.
This is the url passed in:
appsforce.back/mywiki/api.php?action=feedrecentchanges?hidebots=1&hidecategorization=1&hideWikibase=1&translations=filter&urlversion=1&days=7&limit=50&action=pageslist&feedformat=atom

This is what $this->feed_url has as variable after it runs:
http://appsforce.back%2Fmywiki%2Fapi.php/?action=feedrecentchanges?hidebots=1&hidecategorization=1&hideWikibase=1&translations=filter&urlversion=1&days=7&limit=50&action=pageslist&feedformat=atom#

After trying to get the feed I get:
cURL error 6: Could not resolve host: appsforce.back%2Fmywiki%2Fapi.php

**Comment by JanPetterMG** [comment]
You're right, there is an encoding bug in [Misc::fix_protocol](https://github.com/simplepie/simplepie/blob/master/library/SimplePie/Misc.php#L194). Issue #600 submitted.

Anyway @AndreFabris, you may easily work around this issue, until it's fixed, by including the missing protocol in your URL, as mentioned earlier.

Your web browser may not show you the protocol, but that doesn't mean it's not there.
Try this URL instead, I can guarantee you it's working just fine:
[http://appsforce.back/mywiki/api.php?action=feedrecentchanges&hidebots=1&hidecategorization=1&hideWikibase=1&translations=filter&urlversion=1&days=7&limit=50&action=pageslist&feedformat=atom](http://appsforce.back/mywiki/api.php?action=feedrecentchanges?hidebots=1&hidecategorization=1&hideWikibase=1&translations=filter&urlversion=1&days=7&limit=50&action=pageslist&feedformat=atom)
